adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralNativeRenderer.m

```diff
@@ -0,0 +1,266 @@
+    _nativeManager.delegate = self;
+    [_nativeManager loadWithBidToken:adConfiguration.bidResponse];
+}
+
+- (MTGMediaView *)createMediaView {
+    
+    if (_mediaView) {
+        return _mediaView;
+    }
+    
+    _mediaView= [[MTGMediaView alloc] initWithFrame:CGRectZero];
+    _mediaView.delegate = self;
+    return _mediaView;
+}
+
+#pragma mark - MTGBidNativeAdManagerDelegate
+- (void)nativeAdsLoaded:(nullable NSArray *)nativeAds bidNativeManager:(nonnull MTGBidNativeAdManager *)bidNativeManager {
+    
+    if ([nativeAds isKindOfClass:NSArray.class] && nativeAds.count > 0) {
+        _campaign = nativeAds.firstObject;
+        
+        if (![GADMAdapterMintegralUtils isEmpty:_campaign.iconUrl]) {
+            _icon = [self imageWithUrlString:_campaign.iconUrl];
+        }
+                
+        if (![GADMAdapterMintegralUtils isEmpty:_campaign.imageUrl]) {
+            GADNativeAdImage *image = [self imageWithUrlString:_campaign.imageUrl];
+            if (image) {
+                _images = @[image];
+            }
+        }
+        
+        [self createMediaView];
+        
+        if (_adLoadCompletionHandler) {
+            _adEventDelegate = _adLoadCompletionHandler(self,nil);
+        }
+
+    }else{
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorAdNotFill, @"Native ad not fill");
+        if (_adLoadCompletionHandler) {
+            _adLoadCompletionHandler(nil,error);
+        }
+    }
+}
+
+- (void)nativeAdsFailedToLoadWithError:(nonnull NSError *)error bidNativeManager:(nonnull MTGBidNativeAdManager *)bidNativeManager {
+    if (_adLoadCompletionHandler) {
+        _adLoadCompletionHandler(nil,error);
+    }
+}
+
+#pragma mark - MTGBidNativeAdManagerDelegate
+- (void)nativeAdDidClick:(nonnull MTGCampaign *)nativeAd bidNativeManager:(nonnull MTGBidNativeAdManager *)bidNativeManager {
+    [_adEventDelegate reportClick];
+}
+
+- (void)nativeAdImpressionWithType:(MTGAdSourceType)type bidNativeManager:(nonnull MTGBidNativeAdManager *)bidNativeManager {
+    [_adEventDelegate reportImpression];
+}
+
+#pragma mark -MTGMediaViewDelegate
+- (void)MTGMediaViewWillEnterFullscreen:(MTGMediaView *)mediaView {
+    [_adEventDelegate willPresentFullScreenView];
+}
+
+- (void)MTGMediaViewDidExitFullscreen:(MTGMediaView *)mediaView {
+    [_adEventDelegate didDismissFullScreenView];
+}
+
+- (void)MTGMediaViewVideoDidStart:(MTGMediaView *)mediaView {
+    [_adEventDelegate didPlayVideo];
+}
+
+- (void)MTGMediaViewVideoPlayCompleted:(MTGMediaView *)mediaView {
+    [_adEventDelegate didEndVideo];
+}
+
+- (void)nativeAdDidClick:(nonnull MTGCampaign *)nativeAd mediaView:(MTGMediaView *)mediaView {
+    [_adEventDelegate reportClick];
+}
+
+- (void)nativeAdImpressionWithType:(MTGAdSourceType)type mediaView:(MTGMediaView *)mediaView {
+    [_adEventDelegate reportImpression];
+}
```

Comment:
Since we already call the `reportClick` and `reportImpression` above, will they be double-called if we add them here?

Reply:
no

Reply:
- (void)nativeAdImpressionWithType:(MTGAdSourceType)type bidNativeManager:(nonnull MTGBidNativeAdManager *)bidNativeManager
The publisher is not using the mintegral  media view, this method will be called back when the ad is displayed.